src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php

```diff
@@ -290,6 +290,42 @@ protected function stream(string $view, array $parameters = [], StreamedResponse
         return $response;
     }
 
+    /**
+     * Handles a form:
+     *
+     * * if the form is not submitted, the Twig template passed in $view is rendered and a 200 HTTP status code is set
+     * * if the form is submitted but invalid, the Twig template passed in $view is rendered and 422 HTTP status code is set
+     * * if the form is submitted and valid, the entity is saved (only if it is managed by Doctrine ORM), a 306 HTTP status code is set and the Location HTTP header is set to the value of $redirectUrl
+     */
+    public function handleForm(Request $request, FormInterface $form, string $view, ?string $redirectUrl = null, ?callable $onSuccess = null, array $viewParameters = [], ?Response $response = null): Response
```

Comment:
The signature is not friendly to cases where doing a simple persist and flush using the ORM is not enough and you need to pass a `$onSuccess` callable instead, as it still requires passing the unused `$redirectUrl` param before it.

Maybe the basic helper should always take a callable, and then another ORM-specific helper should take a redirectUrl and create a callable flushing and redirecting ?

Reply:
It’s not an issue with PHP8+ thanks to named parameters.

Reply:
except that Symfony does not provide BC yet for the name of parameters (except for attribute constructors)